### PR TITLE
AppVeyor

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

**Comment by lrytz** [comment]
This is ready. I ran a manual build with `mode=release`, `version=2.12.5-bin-3995c7e`, https://travis-ci.org/scala/scala-dist/builds/346991047
  - first it triggered the windows job (https://ci.appveyor.com/project/scala/scala-dist/build/37), which successfully uploaded `scala-2.12.5-bin-3995c7e.msi` to s3
  - the linux job then built and uploaded the other packages (zip/tgz/rpm/deb). the linux job has some `[error]` output, but it's exactly the same as on older logs on jenkins.
  - finally, the linux job triggered the smoketest (https://travis-ci.org/scala/scala-dist-smoketest/builds/346996198), which also passed

For the record, triggering a release with custom env vars is [described in `build.sh`](https://github.com/scala/scala-dist/pull/182/files#diff-3acefdae08499733e855dd23e1af933d).

I also ran the [`archives`](https://travis-ci.org/scala/scala-dist/builds/347019308) and [`update-website`](https://travis-ci.org/scala/scala-dist/builds/347019933) modes, looks good too. We should probably combine them into one.

**Comment by lrytz** [comment]
https://www.scala-lang.org/files/archive/nightly/2.12.x/?C=M;O=D shows the new modified date for the `2.12.5-bin-3995c7e` artifacts

File: build.sbt [hunk 1]
```diff
@@ -20,21 +20,28 @@ mappings in upload := Seq()
...
-  // the publishing job runs on an EC2 instance that has access to the S3 bucket via its IAM instance profile
-  val client = AmazonS3ClientBuilder.standard.withRegion(Regions.US_EAST_1).build 
+  def env(v: String) = {
+    val r = System.getenv(v)
+    assert(r != null, s"Credentials env not specified: $v")
+    r
+  }
+
+  val awsCreds = new BasicAWSCredentials(env("AWS_ACCESS_KEY_ID"), env("AWS_SECRET_ACCESS_KEY"))
+  val client = AmazonS3ClientBuilder.standard
```
Comment (adriaanm) [inline build.sbt]: I'd expect `AmazonS3ClientBuilder.standard`  to read the env variables -- did that not work?

File: admin/build.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#    works in the REST API. We use `before_install` and assume it's not used otherwise.
+#    (*) https://docs.travis-ci.com/user/triggering-builds/#Customizing-the-build-configuration
+#  - Available modes:
+#    - `release` to build native packages and upload them to S3
+#    - `archive` to copy archives to chara (for scala-lang.org)
+#    - `update-api` to update the scaladoc api symlinks
+#    In all of the above modes, the `version` needs to be specified.
+#
+# before_install:
+#   - export version=2.12.4
```
Comment (adriaanm) [inline admin/build.sh]: à propos, can also do this in one line `export version=2.12.4 mode=archives`

File: admin/build.ps1 [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    echo "Last command failed."
+    Exit 1
+  }
+}
+
+# oh boy. i don't (want to) fully understand, but executing commands and redirecting is difficult.
+#  - https://mnaoumov.wordpress.com/2015/01/11/execution-of-external-commands-in-powershell-done-right/
+#  - https://stackoverflow.com/a/35980675/248998
+# letting cmd do the redirect avoids confusing powershell.
+& cmd /c 'java -version' '2>&1'
```
Comment (adriaanm) [inline admin/build.ps1:14]: 😓

**Review by adriaanm — APPROVED** [review]
💯

File: build.sbt (hunk 1 above)
Comment (lrytz) [reply build.sbt]: Good point, I didn't realize. Yes it works.

**Pull request merged**